src/consensus/pbft/libbyz/Replica.h

```diff
@@ -147,9 +150,16 @@ class Replica : public Node, public IMessageReceiveBase
   int my_id() const;
   char* create_response_message(int client_id, Request_id rid, uint32_t size);
 
+  // variables used to keep track of versions so that we can tell the kv to
+  // rollback
+  kv::Version last_te_version = 0;
```

Comment:
What does `te` stand for?

Reply:
tentative execution

Reply:
Would it worth adding comments about the role of each variable?

Reply:
sure